### PR TITLE
Update integration test to latest and use Helm 3

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/11035

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,86 +1,13 @@
...
-      E2E_TEST_DIR: "integration/test/basic"
-      E2E_TEST_NAME: "basic"
-    <<: *e2eTest
-  
-  e2eTestMetrics:
-    environment:
-      E2E_TEST_DIR: "integration/test/metrics"
-      E2E_TEST_NAME: "metrics"
-    <<: *e2eTest
+  architect: giantswarm/architect@0.9.0
```
Comment (rossf7) [inline .circleci/config.yml:3]: Latest architect and using integration test job.

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+require (
+	github.com/giantswarm/appcatalog v0.2.3
+	github.com/giantswarm/backoff v0.2.0
+	github.com/giantswarm/e2esetup v0.3.0
+	github.com/giantswarm/e2etests v0.3.1
+	github.com/giantswarm/helmclient v1.0.2
+	github.com/giantswarm/k8sclient v0.2.0
+	github.com/giantswarm/microerror v0.2.0
+	github.com/giantswarm/micrologger v0.3.1
+	k8s.io/apimachinery v0.17.2
```
Comment (rossf7) [inline go.mod:14]: Using helm 3 so imports are sane.

File: helm/metrics-server-app/templates/priorityclass.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{{- if .Values.e2e }}
```
Comment (rossf7) [inline helm/metrics-server-app/templates/priorityclass.yaml:1]: Previously EnsureTillerInstalled created the priority class now we need this template.

File: helm/metrics-server-app/values.yaml [hunk 4]
```diff
@@ -9,6 +9,8 @@ k8sAppLabel: metrics-server
 
 replicas: 1
 
+e2e: false
```
Comment (stone-z) [inline helm/metrics-server-app/values.yaml:12]: What happens if this is true?

Comment (rossf7) [reply helm/metrics-server-app/values.yaml:12]: It creates the giantswarm-critical priority class. In production this is created with ignition.

With Helm 2 we called EnsureTillerInstalled which needed to do this so the Tiller pod would start.

**Pull request merged**